lib/index.js

```diff
@@ -1233,27 +1282,82 @@ Stream.prototype.pull = function (f) {
  *     // ys will be [1, 2]
  * });
  *
+ * var xs2 = _();
+ * xs.write(1, null, function () {
+ *     console.log('written');
+ * });
+ * xs.end();
+ * xs.toArray(_.log); // 'written' will be printed here.
+ *
  * // Do *not* do this.
- * var xs2 = _().toArray(_.log);
- * xs2.write(1); // This call is illegal.
+ * var xs3 = _().toArray(_.log);
+ * xs3.write(1); // This call is illegal.
  */
 
-Stream.prototype.write = function (x) {
+Stream.prototype.write = function (x, encoding, cb) {
```

Comment:
Here you could make encoding optional:

    if (typeof encoding === 'function') {
        cb = encoding;
        encoding = null;
    }